lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -62,9 +65,46 @@ sub _save_vars($) {
     close($fd);
 }
 
+sub cache_assets {
+    my ($job) = @_;
+    #cache ISO and HDD
+    foreach my $this_asset (sort grep { /^(ISO|HDD)[_]*\d?$/ } keys %{$job->{settings}}) {
+        log_debug("Found $this_asset, caching " . $job->{settings}->{$this_asset});
+        my $asset = get_asset($job, $this_asset, $job->{settings}{$this_asset});
+        symlink($asset, basename($asset)) or die "cannot create link: $asset, $pooldir";
+    }
+}
+
+sub cache_tests {
+
+    my ($shared_cache, $testpoolserver) = @_;
+    no autodie 'system';
+
+    my $start = time;
+    #Do an flock to ensure only one worker is trying to synchronize at a time.
+    my @cmd = qw(flock -E 999);
+    push @cmd, "$shared_cache/needleslock";
+    push @cmd, (qw(rsync -aP), $testpoolserver, qw(--delete --exclude=.git));
```

Comment:
excluding git means isotovideo won't be able to report the tests hash into the log. So we better sync it, it doesn't change more often than the needles itself